File: validate_stack_sequences.java
class Solution {
    public boolean validateStackSequences(int[] pushed, int[] popped) {
        Stack<Integer> st = new Stack<>();
        int i = 0;

        for (int n : pushed) {
            st.push(n);

            while (!st.isEmpty() && st.peek() == popped[i]) {
                st.pop();
                i++;
            }
        }

        return i == popped.length;        
    }
}

File: validate_stack_sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:

        st = []
        i = 0

        for n in pushed:
            st.append(n)

            while st and st[-1] == popped[i]:
                st.pop()
                i += 1
        
        return i == len(popped)
